cordapp-upgrades/README.md

```diff
@@ -0,0 +1,158 @@
+ uploads each attachment to each node that doesn't already have it.)
+ 
+ 4. Check that obligations can be issued between nodes (again, `./gradlew issueBetweenNodes` can be used). It is
+ instructive to show that obligations can still be freely transferred between nodes running different contract versions
+ at this point. To do this, first clear all obligations from the nodes (the easiest way to do this is to settle them all
+ using `./gradlew settleAllObligations`), then do the following:
+  
+   - Issue an obligation from the shell of PartyB, naming PartyA as the lender (`start IssueObligation$Initiator amount:
+   $100, lender: PartyA, anonymous: true`)
+   - On PartyA, do `run vaultQuery contractStateType: net.corda.examples.obligation.contract.Obligation` and note the
+   linear ID of the obligation state.
+   - Transfer the obligation to PartyC: `start TransferObligation$Initiator newLender: PartyC, linearId: <linear ID>, anonymous: true`
+   - Transfer the obligation back, by running the same command on PartyC and naming PartyA as the new lender.
+  
+5. Shut down the network, then upgrade the final node (PartyC) to use the new contract:
+   - `scripts/upgradeNodes.sh workflows v3-can-default PartyB PartyC`
+   - `scripts/upgradeNodes.sh contracts v2-can-default PartyB PartyC`
+  
+6. Adjust the CorDapp configuration to set the new field in the obligation state. (Note that in a real system, this
+ would be done with a new version of the workflows CorDapp that starts to populate the field.) To do this, go to each
+ node's CorDapp configuration (build/nodes/<node>/cordapps/config/obligation-workflows.conf) and set the only field in
+ that file to true.
+ 
+7. Restart the nodes, and try out the commands in step 3 again. Note that the "defaulted" field in the obligation state
+ is set to "false" (rather than null) when querying obligation states using a vault query.
+ 
+8. Try the new defaulting flow, by running `start DefaultObligation$Initiator linearId: <linear ID>` from the borrower
+ of the obligation. This will set the field to true (and nothing else).
+ 
+To see some of the ways this could be set up incorrectly:
+ - Try also setting the configuration option to true for the nodes running the new CorDapp version in step 3. This will
+ prevent the node running the old version (PartyC) from spending the obligation state due to the no downgrade rule. The
+ node will accept the transferred obligation from PartyA, but as there is information in the new field and this would
+ be lost, the transaction to send it back will fail. This means that nodes cannot start using new fields in contract
+ state definitions until all nodes in the network have been upgraded, unless it is acceptable to leave nodes with
+ unspendable states.
+ - To fix the above situation, upgrade the node (follow step 4), then restart the network and try again.
+
+# The Obligation CorDapp
+(Taken from the Obligation CorDapp also in this repo.)
```

Comment:
Why is this included?

Reply:
I thought some background on what the app actually does might be worthwhile, but I can remove this and just point to the original.

Reply:
Actually, I already do point to the original at the start of the README, so I'll just remove this.

Reply:
Remove this section and add link at the beginning of the README as commented above.